File: Assets/AltUnityTester-BindingsAndExamples(can_be_deleted)/Bindings~/java/src/main/java/ro/altom/altunitytester/Commands/ObjectCommand/AltSetText.java
package ro.altom.altunitytester.Commands.ObjectCommand;

import com.google.gson.Gson;
import ro.altom.altunitytester.AltBaseSettings;
import ro.altom.altunitytester.AltUnityObject;
import ro.altom.altunitytester.Commands.AltCommandReturningAltObjects;

public class AltSetText extends AltCommandReturningAltObjects {
    private AltUnityObject altUnityObject;
    private String newText;
    
    public AltSetText(AltBaseSettings altBaseSettings, AltUnityObject altUnityObject, String text) {
        super(altBaseSettings);
        this.altUnityObject = altUnityObject;
        this.newText = text;
    }
    public AltUnityObject Execute() {
        String altObject = new Gson().toJson(altUnityObject);
        send(CreateCommand("setText", altObject, newText));
        return ReceiveAltUnityObject();
    }
}

File: Assets/AltTester/AltDriver/Commands/ObjectCommands/AltSetText.cs
namespace Altom.AltDriver.Commands
{
    public class AltSetText : AltCommandReturningAltElement
    {
        AltSetTextParams cmdParams;

        public AltSetText(IDriverCommunication commHandler, AltObject altUnityObject, string text, bool submit) : base(commHandler)
        {
            cmdParams = new AltSetTextParams(altUnityObject, text, submit);
        }

        public AltObject Execute()
        {
            CommHandler.Send(cmdParams);
            return ReceiveAltObject(cmdParams);
        }
    }
}